gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -273,42 +276,55 @@ private boolean isDirectPathEnabled() {
     return false;
   }
 
+  private boolean isDirectPathXdsEnabledViaBuilderOption() {
+    return Boolean.TRUE.equals(attemptDirectPathXds);
+  }
+
+  private boolean isDirectPathXdsEnabledViaEnv() {
+    String directPathXdsEnv = envProvider.getenv(DIRECT_PATH_ENV_ENABLE_XDS);
+    return Boolean.parseBoolean(directPathXdsEnv);
+  }
+
   @InternalApi
   public boolean isDirectPathXdsEnabled() {
-    // Method 1: Enable DirectPath xDS by option.
-    if (Boolean.TRUE.equals(attemptDirectPathXds)) {
-      return true;
-    }
-    // Method 2: Enable DirectPath xDS by env.
-    String directPathXdsEnv = envProvider.getenv(DIRECT_PATH_ENV_ENABLE_XDS);
-    boolean isDirectPathXdsEnv = Boolean.parseBoolean(directPathXdsEnv);
-    if (isDirectPathXdsEnv) {
-      return true;
-    }
-    return false;
+    return isDirectPathXdsEnabledViaEnv() || isDirectPathXdsEnabledViaBuilderOption();
   }
 
   // This method should be called once per client initialization, hence can not be called in the
   // builder or createSingleChannel, only in getTransportChannel which creates the first channel
   // for a client.
   private void logDirectPathMisconfig() {
     if (isDirectPathXdsEnabled()) {
-      // Case 1: does not enable DirectPath
       if (!isDirectPathEnabled()) {
-        LOG.log(
-            Level.WARNING,
-            "DirectPath is misconfigured. Please set the attemptDirectPath option along with the"
-                + " attemptDirectPathXds option.");
+        // Only two cases are possible: Direct path xDS enabled via env var or via builder.
+        // Case 1: Direct Path is only enabled via XDS env var. We will _warn_ the user that this is
+        // a misconfiguration if they intended to set the env var.
+        if (isDirectPathXdsEnabledViaEnv()) {
+          LOG.log(
+              Level.WARNING,
+              "Env var "
+                  + DIRECT_PATH_ENV_ENABLE_XDS
+                  + " was found and set to TRUE, but DirectPath was not enabled for this client. If this is intended for "
+                  + "this client, please note that this is a misconfiguration and set the attemptDirectPath option as well.");
+        }
+        // Case 2: Direct Path xDS was enabled via Builder. Direct Path TD (via gRPCLB) must be set
+        // along with xDS.
```

Comment:
```suggestion
        // Case 2: Direct Path xDS was enabled via Builder. Direct Path Traffic Director must be set (enabled with `setAttemptDirectPath(true)`) along with xDS.
```

Let me know if this the suggestion makes sense. I just took out the abbreviation TD -> Traffic Director and added that TD can be enabled with `setAttemptDirectPath(true)`